app/Http/Controllers/CityOpinionController.php

```diff
@@ -6,24 +6,21 @@
 
 use App\Http\Requests\CityOpinionRequest;
 use App\Models\CityOpinion;
-use Illuminate\Support\Facades\Auth;
 
 class CityOpinionController extends Controller
 {
     public function store(CityOpinionRequest $request): void
     {
-        $opinion = $request->only(["rating", "content", "city_id"]);
-        $opinion["user_id"] = Auth::id();
-
-        CityOpinion::query()->create($opinion);
+        $request->user()
+            ->cityOpinions()
+            ->create($request->validated());
     }
 
     public function update(CityOpinionRequest $request, CityOpinion $cityOpinion): void
     {
-        $opinion = $request->only(["rating", "content", "city_id"]);
-        $opinion["user_id"] = Auth::id();
-
-        $cityOpinion->update($opinion);
+        $request->user()
+            ->cityOpinions()
+            ->update($request->validated());
     }
```

Comment:
```suggestion
        $cityOpinion
            ->update($request->validated());
    }
```

Reply:
Should be better because u are updating item which u want - specified in method as `CityOpinion $cityOpinion`, in previous behaviour u should get new opinion or error - I think.